src/app/backend/handler/apihandler.go

```diff
@@ -220,22 +243,22 @@ func CreateHTTPAPIHandler(client *clientK8s.Clientset, heapsterClient client.Hea
 
 	apiV1Ws.Route(
 		apiV1Ws.POST("/appdeployment").
-			To(apiHandler.handleDeploy).
+			To(apiHandler.injectApiClient(apiHandler.handleDeploy)).
```

Comment:
Taking a closer look at the code I specifically added this wrapper function so it can handle all the error handling and we get less code duplication. The other approach would look like:

```
func (apiHandler *APIHandler) getApiClient(request *restful.Request) (*clientK8s.Clientset, error) {
	cfg, err := apiHandler.clientConfig.ClientConfig()
	if err != nil {
		return nil, err
	}

	token := request.HeaderParameter("Authorization")
	if strings.HasPrefix(token, "Bearer ") {
		cfg.BearerToken = strings.TrimPrefix(token, "Bearer ")
	}

	apiclient, err := clientK8s.NewForConfig(cfg)
	if err != nil {
		return nil, err
	}

	return apiclient, nil
}
```


```
	apiClient, err := apiHandler.getApiClient(request)
	
	if err != nil {
		handleInternalError(response, err)
		return
	}
```

If you're ok with duplicating the last part about for every endpoint I can make that change.